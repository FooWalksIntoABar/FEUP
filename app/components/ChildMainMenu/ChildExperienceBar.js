import React from 'react'
import { View, StyleSheet, Text, Image, TouchableWithoutFeedback } from 'react-native'
import PropTypes from 'prop-types'

import * as Progress from 'react-native-progress'

import Images from '../../assets/images/images'
import Layout from '../../constants/Layout'
import { accentColor } from '../../styles/Colors'

const barHeight = Layout.window.height * 0.08

export class ChildExperienceBar extends React.Component {
  render () {
    return (
      <View style={styles.expirienceBarContainer}>
        <View style={[styles.progressBarContainer]} >
          <Progress.Bar
            width={null}
            height={barHeight}
            borderRadius={0}
            borderWidth={0}
            progress={this.props.progress}
            color={accentColor}
          />
          <View style={[{ left: '25%' }, styles.progressBarDivider]} />
          <View style={[{ left: '50%' }, styles.progressBarDivider]} />
          <View style={[{ left: '75%' }, styles.progressBarDivider]} />
        </View>
        <View style={styles.experienceBarCircle}>
          <TouchableWithoutFeedback onPress={this.props.onPress}>
            <Image style={styles.rewardImage} source={Images.pool} resizeMode={'cover'} />
          </TouchableWithoutFeedback>
        </View>
        <Text style={styles.levelText}>NÃ­vel {this.props.level}</Text>
      </View>
    )
  }
}

ChildExperienceBar.propTypes = {
  progress: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired,
  onPress: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
  expirienceBarContainer: {
    height: barHeight * 2,
    width: '100%',
    justifyContent: 'flex-end',
    paddingLeft: barHeight,
    paddingRight: barHeight * 1.5
  },
  progressBarContainer: {
    backgroundColor: 'gray',
    height: barHeight,
    width: '100%',
    borderRadius: barHeight / 2,
    elevation: 6,
    overflow: 'hidden',
    position: 'relative'
  },
  progressBarDivider: {
    borderRightColor: '#000',
    borderRightWidth: 2,
    opacity: 0.5,
    height: barHeight,
    position: 'absolute'
  },
  experienceBarCircle: {
    width: barHeight * 1.5,
    height: barHeight * 1.5,
    borderRadius: barHeight * 0.75,
    elevation: 7,
    backgroundColor: 'white',
    right: barHeight * 0.75,
    bottom: barHeight * 0.25,
    position: 'absolute',
    overflow: 'hidden'
  },
  levelText: {
    fontFamily: 'LinotteBold',
    fontSize: 18,
    height: barHeight / 2,
    color: '#fff',
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 3,
    marginLeft: barHeight / 3 + 6
  },
  rewardImage: {
    width: barHeight * 1.5,
    height: barHeight * 1.5
  }
})
