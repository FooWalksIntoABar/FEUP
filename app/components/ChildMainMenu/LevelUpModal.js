import React from 'react'
import { TouchableOpacity, Image, View, Text } from 'react-native'
import PropTypes from 'prop-types'

import Modal from 'react-native-modalbox'
import { RewardSound } from '../RewardsModal/RewardSound'
import styles from '../../styles/ChildMainMenuScreen.style'

import Images from '../../assets/images/images'

export class LevelUpModal extends React.Component {
  state = {
    isOpened: this.props.show
  }

  onPress = () => this.setState({ isOpened: false })

  render () {
    return (
      <View style={styles.modalContainer}>
        <Modal style={styles.levelUpModal} isOpen={this.state.isOpened} onClosed={this.props.onClosed}>
          <View style={styles.levelUpCardTitleContainer}>
            <Text style={styles.levelUpCardTitle}>{this.props.isReward ? `NÃ­vel ${this.props.level - 1} completo!` : `Faltam ${(this.props.level + 1) * 100 - this.props.xp}`}</Text>
            {!this.props.isReward && <Image style={styles.levelUpStarIcon} source={Images.ui.star} resizeMode='contain' />}
            {!this.props.isReward && <Text style={styles.levelUpCardTitle}> para</Text>}
          </View>
          <View style={styles.levelUpCard}>
            <Image style={styles.levelUpRewardImage} source={Images.pool} resizeMode={'cover'} />
            <View style={styles.levelUpCardBottomContainer}>
              <Text style={styles.levelUpText}>Dia na piscina</Text>
              {this.props.isReward && <TouchableOpacity
                activeOpacity={0.8}
                style={styles.levelUpButton}
                onPress={this.onPress}>
                <Image
                  style={styles.levelUpButtonImage}
                  resizeMode={'contain'}
                  source={Images.ui.next} />
              </TouchableOpacity>}
            </View>
          </View>
        </Modal>
        {this.props.isReward && <RewardSound success />}
      </View>
    )
  }
}

LevelUpModal.propTypes = {
  show: PropTypes.bool.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  isReward: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired
}
