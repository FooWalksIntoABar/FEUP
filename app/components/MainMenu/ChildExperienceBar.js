import React from 'react'
import { View, StyleSheet, Text, Image } from 'react-native'
import PropTypes from 'prop-types'

import * as Progress from 'react-native-progress'

import Images from '../../assets/images/images'
import Layout from '../../constants/Layout'

const barHeight = Layout.window.height * 0.08

export class ChildExperienceBar extends React.Component {
  render () {
    return (
      <View style={styles.expirienceBarContainer}>
        <View style={[styles.progressBarContainer]} >
          <Progress.Bar
            width={null}
            height={barHeight}
            borderRadius={0}
            borderWidth={0}
            progress={this.props.progress}
            color={'#700548'}
          />
          <View style={[{ left: '25%' }, styles.progressBarDivider]} />
          <View style={[{ left: '50%' }, styles.progressBarDivider]} />
          <View style={[{ left: '75%' }, styles.progressBarDivider]} />
        </View>
        <View style={styles.experienceBarCircle} >
          <Image style={styles.rewardImage} source={Images.pool} resizeMode={'cover'} />
        </View>
        <Text style={styles.levelText}>NÃ­vel {this.props.level}</Text>
      </View>
    )
  }
}

ChildExperienceBar.propTypes = {
  progress: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired
}

const styles = StyleSheet.create({
  expirienceBarContainer: {
    height: barHeight * 2,
    width: '100%',
    justifyContent: 'flex-end',
    paddingLeft: barHeight,
    paddingRight: barHeight * 1.5
  },
  progressBarContainer: {
    backgroundColor: 'gray',
    height: barHeight,
    width: '100%',
    borderRadius: barHeight / 2,
    elevation: 6,
    overflow: 'hidden',
    position: 'relative'
  },
  progressBarDivider: {
    borderRightColor: '#000',
    borderRightWidth: 2,
    opacity: 0.5,
    height: barHeight,
    position: 'absolute'
  },
  experienceBarCircle: {
    width: barHeight * 1.5,
    height: barHeight * 1.5,
    borderRadius: barHeight * 0.75,
    elevation: 7,
    backgroundColor: 'white',
    right: barHeight * 0.75,
    bottom: barHeight * 0.25,
    position: 'absolute',
    overflow: 'hidden'
  },
  levelText: {
    fontFamily: 'LinotteBold',
    fontSize: 18,
    height: barHeight / 2,
    color: '#000',
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 3,
    marginLeft: barHeight / 3 + 6
  },
  rewardImage: {
    width: barHeight * 1.5,
    height: barHeight * 1.5
  }
})
