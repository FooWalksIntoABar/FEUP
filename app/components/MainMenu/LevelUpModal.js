import React from 'react'
import { TouchableOpacity, Image, View, Text } from 'react-native'
import PropTypes from 'prop-types'

import Modal from 'react-native-modalbox'
import styles from '../../styles/ChildMainMenuScreen.style'

import Images from '../../assets/images/images'

export class LevelUpModal extends React.Component {
  state = {
    isOpened: this.props.show
  }

  render () {
    return (
      <View style={styles.modalContainer}>
        <Modal style={styles.levelUpModal} isOpen={this.state.isOpened} onClosed={this.props.onClosed}>
          <View style={styles.levelUpCard}>
            <Image style={styles.levelUpRewardImage} source={Images.pool} resizeMode={'cover'} />
            <Text style={styles.levelUpText}>Vamos Ã  piscina</Text>
            <TouchableOpacity
              activeOpacity={0.8}
              style={styles.levelUpButton}
              onPress={() => this.setState({ isOpened: false })}>
              <Image
                style={styles.levelUpButtonImage}
                resizeMode={'contain'}
                source={Images.ui.next} />
            </TouchableOpacity>
          </View>
        </Modal>
      </View>
    )
  }
}

LevelUpModal.propTypes = {
  show: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired
}
