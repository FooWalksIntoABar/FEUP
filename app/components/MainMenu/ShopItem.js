import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'

import styles from '../../styles/Shop.style'
import PropTypes from 'prop-types'

import Images from '../../assets/images/images'

export class ShopItem extends React.Component {
  render () {
    return (
      <View style={styles.shopItemContainer} >
        <TouchableOpacity style={this.props.canAfford ? styles.shopItem : styles.shopItemDisabled} activeOpacity={0.6} disabled={!this.props.canAfford}>
          <Image style={[styles.shopItemImage, !this.props.canAfford && { opacity: 0.3 }]} source={Images.cap} recizeMode='contain' />
          <View style={[styles.itemPrice, !this.props.canAfford && { opacity: 0.3 }]} >
            <Image style={styles.currencyIcon} source={Images.ui.star} resizeMode='contain' />
            <Text style={styles.currencyText}>{this.props.cost}</Text>
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

ShopItem.propTypes = {
  cost: PropTypes.number.isRequired,
  canAfford: PropTypes.bool.isRequired
}
