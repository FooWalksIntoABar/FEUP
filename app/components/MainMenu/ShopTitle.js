import React from 'react'
import { View, Text } from 'react-native'

import styles from '../../styles/Shop.style'
import PropTypes from 'prop-types'

export class ShopTitle extends React.Component {
  render () {
    return (
      <View style={styles.shopTitleContainer} >
        <Text style={styles.shopTitle}>{this.props.title}</Text>
      </View>
    )
  }
}

ShopTitle.propTypes = {
  title: PropTypes.string.isRequired
}
