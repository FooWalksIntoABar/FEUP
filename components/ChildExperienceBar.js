import React from 'react'
import { View, StyleSheet, Text } from 'react-native'
import PropTypes from 'prop-types'

import * as Progress from 'react-native-progress'

import Layout from '../constants/Layout'

export class ChildExperienceBar extends React.Component {
  render () {
    return (
      <View>
        <View style={[styles.progressBarContainer]} >
          <Progress.Bar
            width={null}
            height={Layout.window.height * 0.12 / 1.5}
            borderRadius={0}
            borderWidth={0}
            progress={this.props.progress}
          />
          <View style={[{ left: '25%' }, styles.progressBarDivider]} />
          <View style={[{ left: '50%' }, styles.progressBarDivider]} />
          <View style={[{ left: '75%' }, styles.progressBarDivider]} />
        </View>
        <View style={styles.experienceBarCircle} >
          <Text>{this.props.level}</Text>
        </View>
      </View>
    )
  }
}

ChildExperienceBar.propTypes = {
  progress: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired
}

const styles = StyleSheet.create({
  progressBarContainer: {
    flexGrow: 1,
    backgroundColor: 'gray',
    marginRight: Layout.window.height * 0.03 / 2,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.27,
    shadowRadius: 4.65,
    elevation: 6,
    height: Layout.window.height * 0.12 / 1.5,
    borderRadius: Layout.window.height * 0.12 / 3,
    overflow: 'hidden',
    position: 'relative'
  },
  progressBarDivider: {
    borderRightColor: '#000',
    borderRightWidth: 2,
    opacity: 0.5,
    height: Layout.window.height * 0.12 / 1.5,
    position: 'absolute'
  },
  experienceBarCircle: {
    width: Layout.window.height * 0.1,
    height: Layout.window.height * 0.1,
    borderRadius: Layout.window.height * 0.1 / 2,
    backgroundColor: 'white',
    borderColor: '#9999FF',
    borderWidth: 3,
    alignItems: 'center',
    justifyContent: 'center',
    left: Layout.window.width * 1.15,
    top: -Layout.window.height * 0.01,
    position: 'absolute',
    elevation: 7,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.27,
    shadowRadius: 4.65
  }
})
