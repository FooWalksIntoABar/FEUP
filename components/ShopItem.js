import React from 'react'
import { View, StyleSheet, Text, Image, TouchableOpacity } from 'react-native'

import Layout from '../constants/Layout'
import PropTypes from 'prop-types'

export class ShopItem extends React.Component {
  render () {
    return (
      <View>
        <TouchableOpacity>
          <View style={styles.shopItem} />
          <View style={styles.itemPrice} >
            <Image style={styles.currencyIcon} source={require('../assets/images/yellow_star.png')} resizeMode='contain' />
            <Text>{this.props.cost}</Text>
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

ShopItem.propTypes = {
  cost: PropTypes.number.isRequired
}

const styles = StyleSheet.create({
  shopItem: {
    width: Layout.window.width * 0.25,
    height: Layout.window.height * 0.2,
    borderColor: 'black',
    borderWidth: 2,
    backgroundColor: '#C5CAE9',
    position: 'relative'
  },
  currencyIcon: {
    width: Layout.window.width * 0.03,
    height: Layout.window.height * 0.03,
    marginHorizontal: Layout.window.width * 0.01
  },
  itemPrice: {
    width: Layout.window.width * 0.25,
    height: Layout.window.height * 0.04,
    position: 'absolute',
    bottom: 0,
    backgroundColor: 'rgba(0, 0, 0, 0.2)',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row'
  }
})
