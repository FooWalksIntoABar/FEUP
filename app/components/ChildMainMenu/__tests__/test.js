import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { ChildExperienceBar } from '../ChildExperienceBar'
import { ShopItem } from '../ShopItem'
import { ShopTitle } from '../ShopTitle'
import { LevelUpModal } from '../LevelUpModal'

configure({ adapter: new Adapter() })

describe('MainMenu components', () => {
  it('renders ChildExperienceBar correctly', () => {
    const wrapper = shallow(<ChildExperienceBar
      progress={0.5}
      level={2}
      onPress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders ShopItem correctly', () => {
    const wrapper = shallow(<ShopItem
      cost={100}
      disabled={false}
      purchased={false}
      purchaseItem={jest.fn()}
      id={0} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ disabled: true })
    wrapper.setProps({ purchased: true })
    wrapper.instance().purchaseItem()
    wrapper.unmount()
  })

  it('renders ShopTitle correctly', () => {
    const wrapper = shallow(<ShopTitle
      title={'test'} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders LevelUpModal correctly', () => {
    const wrapper = shallow(<LevelUpModal
      show={false}
      level={3}
      xp={350}
      isReward={false}
      onClosed={jest.fn()}
      playSounds={false} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ isReward: true })
    wrapper.instance().onPress()
    wrapper.unmount()
  })
})
