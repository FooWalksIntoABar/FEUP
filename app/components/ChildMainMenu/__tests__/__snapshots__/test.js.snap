// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu components renders ChildExperienceBar correctly 1`] = `
<Component
  style={
    Object {
      "height": 213.44,
      "justifyContent": "flex-end",
      "paddingLeft": 106.72,
      "paddingRight": 160.07999999999998,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "backgroundColor": "gray",
          "borderRadius": 53.36,
          "elevation": 6,
          "height": 106.72,
          "overflow": "hidden",
          "position": "relative",
          "width": "100%",
        },
      ]
    }
  >
    <ProgressBar
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      borderRadius={0}
      borderWidth={0}
      color="#a4e76f"
      height={106.72}
      indeterminate={false}
      progress={0.5}
      useNativeDriver={false}
      width={null}
    />
    <Component
      style={
        Array [
          Object {
            "left": "25%",
          },
          Object {
            "borderRightColor": "#000",
            "borderRightWidth": 2,
            "height": 106.72,
            "opacity": 0.5,
            "position": "absolute",
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "left": "50%",
          },
          Object {
            "borderRightColor": "#000",
            "borderRightWidth": 2,
            "height": 106.72,
            "opacity": 0.5,
            "position": "absolute",
          },
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "left": "75%",
          },
          Object {
            "borderRightColor": "#000",
            "borderRightWidth": 2,
            "height": 106.72,
            "opacity": 0.5,
            "position": "absolute",
          },
        ]
      }
    />
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 80.03999999999999,
        "bottom": 26.68,
        "elevation": 7,
        "height": 160.07999999999998,
        "overflow": "hidden",
        "position": "absolute",
        "right": 80.03999999999999,
        "width": 160.07999999999998,
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[MockFunction]}
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Object {
            "height": 160.07999999999998,
            "width": 160.07999999999998,
          }
        }
      />
    </TouchableWithoutFeedback>
  </Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "LinotteBold",
        "fontSize": 18,
        "height": 53.36,
        "marginLeft": 41.57333333333333,
        "textShadowColor": "rgba(0, 0, 0, 0.2)",
        "textShadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "textShadowRadius": 3,
      }
    }
  >
    NÃ­vel 
    2
  </Text>
</Component>
`;

exports[`MainMenu components renders LevelUpModal correctly 1`] = `
<Component
  style={
    Object {
      "elevation": 99,
      "height": "100%",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <ModalBox
    animationDuration={400}
    backButtonClose={false}
    backdrop={true}
    backdropColor="black"
    backdropContent={null}
    backdropOpacity={0.5}
    backdropPressToClose={true}
    coverScreen={false}
    easing={[Function]}
    isOpen={false}
    keyboardTopOffset={22}
    onClosed={[MockFunction]}
    position="center"
    startOpen={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
    swipeThreshold={50}
    swipeToClose={true}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#bbecec",
          "borderRadius": 50.025000000000006,
          "elevation": 12,
          "flexDirection": "row",
          "height": 100.05000000000001,
          "justifyContent": "center",
          "left": 266.8,
          "paddingHorizontal": 50.025000000000006,
          "position": "absolute",
          "top": 83.375,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#061111",
            "fontFamily": "LinotteBold",
            "fontSize": 66.7,
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "textShadowRadius": 1,
          }
        }
      >
        Faltam 50
      </Text>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 60.03,
            "width": 60.03,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#061111",
            "fontFamily": "LinotteBold",
            "fontSize": 66.7,
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "textShadowRadius": 1,
          }
        }
      >
         para
      </Text>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3db3e2",
          "borderRadius": 33.35,
          "elevation": 6,
          "flexGrow": 1,
          "marginHorizontal": 200.10000000000002,
          "marginVertical": 133.4,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
      <Component
        style={
          Object {
            "alignItems": "flex-end",
            "bottom": 0,
            "flexDirection": "row",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "flex": 1,
              "fontFamily": "LinotteBold",
              "fontSize": 66.7,
              "padding": 40.019999999999996,
              "textShadowColor": "rgba(0, 0, 0, 0.6)",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 5,
            }
          }
        >
          Dia na piscina
        </Text>
      </Component>
    </Component>
  </ModalBox>
</Component>
`;

exports[`MainMenu components renders ShopItem correctly 1`] = `
<Component
  style={
    Object {
      "aspectRatio": 1,
      "padding": 18.75,
      "width": "33.3333%",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.6}
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#bbecec",
        "borderRadius": 18.75,
        "elevation": 6,
        "height": "100%",
        "justifyContent": "center",
        "paddingBottom": 56.25,
        "paddingHorizontal": 9.375,
        "paddingTop": 9.375,
        "width": "100%",
      }
    }
  >
    <Image
      recizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "aspectRatio": 1,
            "height": "100%",
          },
          false,
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 9.375,
            "flexDirection": "row",
            "justifyContent": "center",
            "position": "absolute",
          },
          false,
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 37.5,
            "marginRight": 9.375,
            "width": 37.5,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "LinotteBold",
            "fontSize": 30,
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 3,
          }
        }
      >
        100
      </Text>
    </Component>
  </TouchableOpacity>
</Component>
`;

exports[`MainMenu components renders ShopTitle correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 75,
      "justifyContent": "space-between",
      "marginBottom": 18.75,
      "paddingHorizontal": 18.75,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "LinotteBold",
        "fontSize": 56.25,
        "textShadowColor": "rgba(0, 0, 0, 0.2)",
        "textShadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "textShadowRadius": 3,
      }
    }
  >
    test
  </Text>
</Component>
`;
